### PR TITLE
Increase code coverage in botbuilder library - TeamsActivityHandler class

### PR DESCRIPTION
**Pull request by denscollo**
Addresses #2432

## Description
This PR adds unit tests for the [TeamsActivityHandler](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder/src/teamsActivityHandler.ts#L33) class increasing the code [coverage](https://coveralls.io/builds/31822744/source?filename=libraries/botbuilder/src/teamsActivityHandler.ts) from **73.68%** to **100%**.

## Specific Changes
- Updated the [teamsActivityHandler.test.js](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder/tests/teamsActivityHandler.test.js#L15) file with new unit tests.

## Testing
The following images show the new tests passing.
imagen
imagen

### CODE REVIEW

File: libraries/botbuilder/tests/teamsActivityHandler.test.js [hunk 1]
```diff
@@ -61,6 +61,226 @@ describe('TeamsActivityHandler', () => {
...
+            handleTeamsMessagingExtensionFetchTask(context, action) {
+                return { status: 501 };
+            }
+
+            handleTeamsMessagingExtensionConfigurationQuerySettingUrl(context, action) {
+                return { status: 501 };
+            }
+
+            handleTeamsMessagingExtensionConfigurationSetting(context, settings) {
+                return { status: 501 };
```
Comment (stevengum) [inline libraries/botbuilder/tests/teamsActivityHandler.test.js:95]: If all you're trying to do is not trigger the thrown error from the default implementation, then you don't need to return anything.

Returning `{ status: 501 };` is confusing.

File: libraries/botbuilder/tests/teamsActivityHandler.test.js [hunk 2]
```diff
@@ -61,6 +61,226 @@ describe('TeamsActivityHandler', () => {
...
+            adapter.send(activity)
+                .assertReply(activity => {
+                    assert.strictEqual(activity.value.status, 200, 'should be status code 200.')
+                    done();
+                })
+                .catch(err => done(err));
+        })
+
+        it('should throw an error when onInvokeActivity param context is null.', done => {
+            class InvokeActivity2 extends InvokeActivity{
```
Comment (stevengum) [inline libraries/botbuilder/tests/teamsActivityHandler.test.js:257]: This is another poor name and the inheritance from a subclass of TeamsActivityHandler opens the door for bugs in the test code if someone else modifies `InvokeActivity`.

Please rename the test class and extend TeamsActivityHandler, not the previous subclass.

File: libraries/botbuilder/tests/teamsActivityHandler.test.js [hunk 3]
```diff
@@ -285,109 +523,235 @@ describe('TeamsActivityHandler', () => {
...
+            const taskSubmitActivity = createInvokeActivity('composeExtension/queryLink');
+            adapter.send(taskSubmitActivity)
                 .assertReply(activity => {
-                    assert(activity.type === 'invokeResponse', `incorrect activity type "${activity.type}", expected "invokeResponse"`);
-                    assert(activity.value.status === 200, `incorrect status code "${activity.value.status}", expected "200"`);
-                    assert(!activity.value.body,
-                        `expected empty body for invokeResponse from fileConsent flow.\nReceived: ${JSON.stringify(activity.value.body)}`);
+                    assert.strictEqual(activity.type, 'invokeResponse');
+                    assert.strictEqual(activity.value.status, 501);
+                    assert.strictEqual(activity.value.body, undefined);
```
Comment (stevengum) [inline libraries/botbuilder/tests/teamsActivityHandler.test.js:539]: Thank you for cleaning this up! ❤️

**Comment by denscollo** [comment]
@stevengum  Thanks for the feedback, we just pushed the changes.

**Pull request merged**